packages/scm/src/browser/scm-widget.tsx

```diff
@@ -130,21 +104,26 @@ export class ScmWidget extends ReactWidget implements StatefulWidget {
         if (!this.isAttached || !this.isVisible) {
             return;
         }
-        this.onRender.push(Disposable.create(() => async () => {
-            const selected = this.node.getElementsByClassName(SELECTED_CLASS)[0];
-            if (selected) {
-                ElementExt.scrollIntoViewIfNeeded(this.node, selected);
-            }
-        }));
+        MessageLoop.sendMessage(this.commitWidget, msg);
+        MessageLoop.sendMessage(this.resourceWidget, msg);
+        MessageLoop.sendMessage(this.amendWidget, msg);
         super.onUpdateRequest(msg);
     }
 
-    protected addScmListKeyListeners = (id: string) => {
-        const container = document.getElementById(id);
-        if (container) {
-            this.addScmListNavigationKeyListeners(container);
-        }
-    };
+    protected onAfterAttach(msg: Message): void {
+        this.node.appendChild(this.commitWidget.node);
+        this.node.appendChild(this.resourceWidget.node);
+        this.node.appendChild(this.amendWidget.node);
+
+        super.onAfterAttach(msg);
+        this.update();
+    }
+
+    protected onActivateRequest(msg: Message): void {
+        super.onActivateRequest(msg);
+        this.commitWidget.focus();
+        this.update();
```

Comment:
we should not render on focus, by that moment the widget already should be attached and visible

Reply:
The extraneous update is now removed.